docs/post-processors/vagrant-registry.mdx

```diff
@@ -0,0 +1,292 @@
+  the HCP Vagrant Box registry
+page_title: HCP Vagrant Box Registry - Post-Processors
+nav_title: HCP Vagrant Box Registry
+---
+
+# HCP Vagrant Box Registry Post-Processor
+
+Type: `vagrant-registry`
+Artifact BuilderId: `hashicorp.post-processor.vagrant-registry`
+
+[HCP Vagrant Box Registry](https://portal.cloud.hashicorp.com/vagrant/discover)
+hosts and serves boxes to Vagrant, allowing you to version and distribute boxes
+to an organization or the public in a simple way.
+
+The Vagrant Registry post-processor enables the upload of Vagrant boxes to HCP
+Vagrant Box Registry. Currently, the Vagrant Registry post-processor will accept
+and upload boxes supplied to it from the [Vagrant](/docs/post-processors/vagrant.mdx) or
+[Artifice](https://developer.hashicorp.com/packer/docs/post-processors/artifice) post-processors and the
+[Vagrant](/docs/builders/vagrant.mdx) builder.
+
+## Workflow
+
+It's important to understand the workflow that using this post-processor
+enforces in order to take full advantage of Vagrant and HCP Vagrant Box Registry.
+
+The use of this processor assume that you currently distribute, or plan to
+distribute, boxes via HCP Vagrant Box Registry. It also assumes you create
+Vagrant Boxes and deliver them to your team in some fashion.
+
+Here is an example workflow:
+
+1. You use Packer to build a Vagrant Box for the `virtualbox` provider
+1. The `vagrant-registry` post-processor is configured to point to the box
+    `hashicorp/foobar` on HCP Vagrant Box Registry via the `box_tag` configuration
+1. The post-processor receives the box from the `vagrant` post-processor
+1. It then creates the box name, or verifies the existence of it, on HCP
+    Vagrant Box Registry
+1. It then creates the configured version, or verifies the existence of it
+1. It then creates the configured provider, or verifies the existence of it
+1. It then creates the configured architecture, or verifies the existence of it
+1. The box artifact is uploaded to HCP Vagrant Box Registry
+1. The upload is verified
+1. The version is released and available to users of the box
+
+## Configuration
+
+The configuration allows you to specify the target box that you have access to
+on HCP Vagrant Box Registry, as well as authentication and version information.
+
+### Required
+
+- `box_tag` (string) - The shorthand tag for your box that maps to the box registry
+  name and the box name. For example, `hashicorp/precise64` is the shorthand tag
+  for the `precise64` box in the `hashicorp` box registry.
+
+- `version` (string) - The version number, typically incrementing a previous
+  version. The version string is validated based on [Semantic
+  Versioning](http://semver.org/). The string must match a pattern that could
+  be semver, and doesn't validate that the version comes after your previous
+  versions.
+
+- `client_id` (string) - The service principal client ID for the HCP API. This 
```

Comment:
Packer relies on these environment variables for connecting to HCP Packer. Normally, a user will set HCP_CLIENT_ID, HCP_CLIENT_SECRET, and HCP_PROJECT_ID when publishing an artifact to HCP Packer or reading an image from HCP Packer. 

Vagrant doesn't have support for HCP project ID but if this post-process was used in environment where these envs are defined would the value of Project ID affect authentication? 

I think if using an org level service principal that can pull an image from HCP Packer to use as the source for VMware that then uses Vagrant to create a box, and publish to vagrant registry inside a different project. The user would need the ability to set a project ID. 

If using project level sec principles then they would still need to provide a project id if HCP Packer registry and Vagrant registry are not in the same project.  

project_id should be added as an optional configuration argument.

Reply:
The organization/project values don't have any effect on the vagrant service API being used here. There is a limited set of API end points that use organization/project values, and they are outside of the scope of any interactions this plugin would have with the service (they are specifically related to the top-level registry management tasks). 

The SDK will apply the organization/project values to requests if they are unset and the requests expect those values. Since none of the requests here use them, even if the values are set they will never be used. I'm happy to add them as config options but any provided values would not be used (or enforced).

Reply:
Got it. No need to add the config. I thought it might be a problem. I confirmed all works as expected.